python_src/src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -469,107 +469,275 @@ def update_directions(db_manager: DatabaseManager) -> None:
     process_logger.log_complete()
 
 
-def update_canonical_stop_sequence(db_manager: DatabaseManager) -> None:
+def update_stop_sequence(db_manager: DatabaseManager) -> None:
     """
-    Update canonical_stop_sequence from static_route_patterns
+    Update canonical_stop_sequence  and sync_stop_sequence from static_route_patterns
     """
-    select_update_params = sa.select(
-        TempEventCompare.service_date,
-        TempEventCompare.static_version_key,
-    ).distinct()
-
-    for result in db_manager.select_as_list(select_update_params):
-        service_date = result["service_date"]
-        version_key = result["static_version_key"]
+    static_canon = (
+        sa.select(
+            StaticRoutePatterns.direction_id,
+            StaticTrips.trunk_route_id,
+            sa.func.coalesce(
+                StaticTrips.branch_route_id, StaticTrips.trunk_route_id
+            ).label("route_id"),
+            StaticStops.parent_station,
+            StaticStopTimes.stop_sequence,
+            StaticRoutePatterns.static_version_key,
+        )
+        .select_from(StaticRoutePatterns)
+        .join(
+            StaticTrips,
+            sa.and_(
+                StaticRoutePatterns.representative_trip_id
+                == StaticTrips.trip_id,
+                StaticRoutePatterns.static_version_key
+                == StaticTrips.static_version_key,
+            ),
+        )
+        .join(
+            StaticStopTimes,
+            sa.and_(
+                StaticRoutePatterns.representative_trip_id
+                == StaticStopTimes.trip_id,
+                StaticRoutePatterns.static_version_key
+                == StaticStopTimes.static_version_key,
+            ),
+        )
+        .join(
+            StaticStops,
+            sa.and_(
+                StaticStopTimes.stop_id == StaticStops.stop_id,
+                StaticStopTimes.static_version_key
+                == StaticStops.static_version_key,
+            ),
+        )
+        .where(
+            StaticRoutePatterns.static_version_key
+            == sa.func.any(
+                sa.func.array(
+                    sa.select(TempEventCompare.static_version_key)
+                    .distinct()
+                    .scalar_subquery()
+                )
+            ),
+            StaticRoutePatterns.route_pattern_typicality == 1,
+        )
+        .subquery("static_canon")
+    )
```

Comment:
this is roughly the same as the `static_sub` that was here before, but using multiple static version keys and adding in the `trunk_route_id` field?

Reply:
Correct, I did some profiling on the query and using this:
```
      StaticRoutePatterns.static_version_key
      == sa.func.any(
          sa.func.array(
              sa.select(TempEventCompare.static_version_key)
              .distinct()
              .scalar_subquery()
          )
      ),
```

Hits the index for `static_version_key`, instead of calling each `static_version_key` separately with a `where =`